### PR TITLE
Add update sub-command

### PR DESCRIPTION
**Pull request by adriendelsalle**
Description
---

Add `update` sub-command.
Reuse init parser in `init`, `create` and `update`.

### CODE REVIEW

File: src/micromamba/main.cpp [hunk 1]
```diff
@@ -684,6 +687,13 @@ install_specs(const std::vector<std::string>& specs, bool create_env = false, in
     trans.execute(prefix_data, pkgs_dirs);
 }
 
+void
+update_specs(const std::vector<std::string>& specs)
+{
+    std::cout << specs[0] << std::endl;
```
Comment (wolfv) [inline src/micromamba/main.cpp:693]: ?

Comment (adriendelsalle) [reply src/micromamba/main.cpp:693]: sorry for that it's wip :)
I should have created a draft PR

Comment (wolfv) [reply src/micromamba/main.cpp:693]: no worries.

**Comment by adriendelsalle** [comment]
It should be ok now @wolfv !

File: src/micromamba/main.cpp [hunk 2]
```diff
@@ -1272,26 +1328,14 @@ parse_file_options()
...
-        "--no-pin", create_options.no_pin, "Ignore pinned packages (from config or prefix file)");
+    init_install_global_parser(subcom);
 
     subcom->add_option(
         "--extra-safety-checks", create_options.extra_safety_checks, "Perform extra safety checks");
 
-    init_network_parser(subcom);
-    init_channel_parser(subcom);
-    init_global_parser(subcom);
-
```
Comment (wolfv) [inline src/micromamba/main.cpp:1294]: you tested that thsi can all be removed, right?

File: src/micromamba/main.cpp [hunk 3]
```diff
@@ -684,6 +692,34 @@ install_specs(const std::vector<std::string>& specs, bool create_env = false, in
...
+            throw std::runtime_error(
+                "No active target prefix.\n\nRun $ micromamba activate <PATH_TO_MY_ENV>\nto activate an environment.\n");
+        }
+
+        PrefixData prefix_data(ctx.target_prefix);
+        prefix_data.load();
+
+        for (const auto& package : prefix_data.m_package_records)
+        {
+            auto name = package.second.name;
```
Comment (wolfv) [inline src/micromamba/main.cpp:712]: auto& is better ;)

**Comment by wolfv** [comment]
looks good! happy to merge, two minor comments.

File: src/micromamba/main.cpp (hunk 2 above)
Comment (adriendelsalle) [reply src/micromamba/main.cpp:1294]: all the content is now in `init_install_global_parser` to shared the code with `install` and `update`

File: src/micromamba/main.cpp (hunk 3 above)
Comment (adriendelsalle) [reply src/micromamba/main.cpp]: done!

**Pull request merged**